doc/source/ray-core/scheduling/ray-oom-prevention.rst

```diff
@@ -0,0 +1,234 @@
+Ray Out-Of-Memory Prevention
+============================
+
+If application tasks or actors consume a large amount of heap space, it can cause the node to run out of memory (OOM). When that happens, the operating system will start killing worker or raylet processes, disrupting the application. OOM may also stall metrics and if this happens on the head node, it may stall the :ref:`dashboard <ray-dashboard>` or other control processes and cause the cluster to become unusable.
+
+In this section we will go over:
+
+- What is the memory monitor and how it works
+
+- How to enable and configure it
+
+- How to use the memory monitor to detect and resolve memory issues
+
+What is the memory monitor?
+---------------------------
+
+The memory monitor is a component that runs on each node in your Ray cluster and is instantiated with a configured threshold.
+The memory monitor periodically checks the memory usage of the different Ray processes on the node,
+which includes the worker heap, the object store, and the raylet as described in :ref:`memory management <memory>`.
+
+The memory monitor is a component that runs within the raylet process on each node. It periodically checks the memory usage, which includes the worker heap, the object store, and the raylet as described in :ref:`memory management <memory>`. If the combined usage exceeds a configurable threshold the raylet will kill a task or actor process to free up memory and prevent Ray from failing.
+
+.. note::
+
+    The memory monitor is in :ref:`alpha <api-stability-alpha>`. It is disabled by default and needs to be enabled by setting the environment variable ``RAY_memory_monitor_interval_ms`` to a value greater than zero when Ray starts. It is available on Linux and is tested with Ray running inside a container that is using cgroup v1. If you encounter issues when running the memory monitor outside of a container or the container is using cgroup v2, please :ref:`file an issue or post a question <oom-questions>`.
```

Comment:
Maybe this should be on the top of the doc?

Reply:
Seems this is the top already for discussing memory monitor?